### ISSUE TITLE
flake8 does not raise any errors

### ISSUE THREAD

**Issue by sobolevn** [mschwager/cohesion#11]
For some reason `flake8` does not produce any warnings. Even if it should.

Reproduction: 
1. create `example.py` from the readme
2. run `flake8 --select=H --cohesion-below=95 example.py`

This should raise one error, since the same file produces this output:

```
» cohesion --below 95 -f example.py
File: example.py
  Class: ExampleClass1 (1:0)
    Function: func1 2/3 66.67%
    Function: func2 1/3 33.33%
    Function: func3 staticmethod
    Total: 33.33%
```

But, nothing happens.
Versions I am using: 

```
{
  "dependencies": [
    {
      "dependency": "setuptools",
      "version": "39.0.1"
    }
  ],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.6.5",
    "system": "Darwin"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "cohesion",
      "version": "0.9.1"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.3.1"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "1.6.0"
    }
  ],
  "version": "3.5.0"
}
```

**Comment by mschwager** [comment]
Currently the `cohesion` `flake8` plugin is disabled by default:

```python
class CohesionChecker(object):
    name = cohesion.__name__
    version = cohesion.__version__
    off_by_default = True
    ...
```

This means you have to enable the extension when using it:

```
$ flake8 --enable-extension=H60 --select=H --cohesion-below=95 example.py
example.py:1:1: H601 class has low (33.33%) cohesion
```

I would consider flipping this, though, so that it's enabled by default. After all, if you've installed the `cohesion` plugin then it's probably fair to say that you want it enabled. Perhaps this would be a good change to make for releasing `1.0.0`?

Thoughts?

**Comment by sobolevn** [comment]
@mschwager sure! Why would not you want to use it?

It is always possible to:
1. disable the plugin completely via the config or cli
2. ignore `H` errors for some specific files

Do you need any help in releasing a `1.0.0` version?

**Comment by mschwager** [comment]
One reason I had in mind for not enabling `cohesion` by default was the possibility of including it by default in `flake8` - similar to [mccabe](http://flake8.pycqa.org/en/latest/manpage.html#description). Both `mccabe` and `cohesion` provide useful feedback on code structure, but I wouldn't say they're as objective as other `flake8` findings (e.g. unused imports), and thus could be turned off by default.

As for your issue with `ConfigurationOptions`, I can see a few options. Your use case seems reasonable and common enough, especially with its similarities to Python 3.7's [dataclasses](https://docs.python.org/3/library/dataclasses.html).

One option would be for `cohesion` to detect classes decorated with `dataclass` and ignore them. This wouldn't help your particular use case (unless you can move towards a `dataclass`), but is an easy way to detect classes that do nothing but hold data. Otherwise how would `cohesion` detect a class like this?

Another option would be for you to use [`flake8`'s `noqa` functionality](http://flake8.pycqa.org/en/latest/user/violations.html#in-line-ignoring-errors) on this class. This seems like a perfectly reasonable exception to the `cohesion` rules, and exactly what `noqa` is there for.

As for a `1.0.0` release, I think I have everything I need to do the release. I just need to mull over some of these more philosophical points before making any changes. I appreciate your offer though!

**Comment by sobolevn** [comment]
@mschwager ignoring classes is a way to go if you would ask me.

I can provide two more examples that should be ignored: `attrs` and `Enum`. They will both have 0 cohesion.

**Comment by mschwager** [comment]
Fixed by fd5fc59.
